src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -685,25 +684,16 @@ def _fix_pandas_df_integer(
             and column_metadata.scale == 0
             and not str(pandas_dtype).startswith("int")
         ):
-            # pandas.to_numeric raises "RuntimeWarning: invalid value encountered in cast"
-            # when it tried to downcast and loses precision. In this case, we try to convert to
-            # int64 using astype() and fallback to to_numeric if we were unsuccessful.
-            with warnings.catch_warnings(record=True) as warning_list:
-                pd_col_with_numeric_downcast = pandas.to_numeric(
+            try:
+                if column_metadata.precision > 10:
```

Comment:
out of curiosity, why is number 10, is this documented somewhere?

Reply:
given that scale=0, precision greater than 10 implies that the number of digits in the number will be greater than 10. max number of digits in int32 is equal to 10 therefore we only convert to int64 when precision info by GS is greater than 10. If we keep using pandas to_numeric for large numbers, it loses precision for larger numbers.

Reply:
aha I see, this is knowledgeable, can we document this in the code comment?

Reply:
How about when the precision is great than int64?

Reply:
If the number cannot fit in int64 - it will throw an OverflowError and we will resort to old behavior to `to_numeric`. In this case, we will have float types instead of integer types. I guess I could tighten the condition and say when precision > 10 and < 20.

```
>>> df = pd.DataFrame({'A': [1, '8'*25]})
>>> df['A'].astype('int64')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    df['A'].astype('int64')
.
.
.
OverflowError: Python int too large to convert to C long
```

Reply:
Update here: We can't have the <20 condition here because GS would not guarantee such tight bounds here.

The use-case for this function is to correct mislabeled integer types back to integers so we should ideally not be having overflows. In case of overflow however, pandas.to_numeric will convert incoming data into a float type and we will move back to the old behavior.